src/pinned_memory_manager.cc

```diff
@@ -192,6 +203,17 @@ PinnedMemoryManager::FreeInternal(void* ptr)
                      << "pinned memory deallocation: "
                      << "addr " << ptr;
       memory_info_.erase(it);
+
+      {
+        std::lock_guard<std::mutex> lk(alloc_info_mtx_);
```

Comment:
Similarly I think this mutex can be moved within the `if (ix != allocated_memory_info_.end())` block.